autogpt_platform/backend/backend/copilot/graphiti/client.py

```
@@ -75,6 +76,50 @@ def derive_group_id(user_id: str) -> str:
+        raise ValueError(
+            f"expert_id '{expert_id[:32]}...' yields empty group_id after sanitization"
+        )
+    if safe_expert_id != expert_id:
+        raise ValueError(
+            "expert_id contains invalid characters for group_id derivation "
+            f"(original length={len(expert_id)}, "
+            f"sanitized='{safe_expert_id[:32]}'). "
+            "Only [a-zA-Z0-9_-] are allowed."
+        )
+
+    scope_digest = hashlib.sha256(f"{user_id}\0{expert_id}".encode("utf-8")).hexdigest()
```

Ruff (0.16.1)

[warning] Unnecessary UTF-8 `encoding` argument to `encode`

Remove unnecessary `encoding` argument

(UP012)